core/src/test/java/org/testcontainers/containers/GenericContainerTest.java

```diff
@@ -0,0 +1,63 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.command.InspectContainerResponse.ContainerState;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.experimental.FieldDefaults;
+import org.apache.commons.io.FileUtils;
+import org.junit.Test;
+import org.rnorth.ducttape.unreliables.Unreliables;
+import org.testcontainers.containers.wait.strategy.AbstractWaitStrategy;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class GenericContainerTest {
+
+    @Test
+    public void shouldReportOOMAfterWait() {
+        try (
+            GenericContainer container = new GenericContainer<>()
+                .waitingFor(new WaitForState(ContainerState::getOOMKilled))
+                .withCreateContainerCmdModifier(it -> {
+                    it.getHostConfig().withMemory(4 * FileUtils.ONE_MB);
+                })
+                .withCommand("sh", "-c", "usleep 100; dd if=/dev/urandom bs=128GB count=1 > test.txt")
```

Comment:
Gulp... Better get this right 😅 😂 

At first sight I was perplexed as to why writing to disk would cause OOMKill at all, but realised that actually the only thing that matters here is having the blocksize be >4MB, as dd buffers into memory before writing.

If that's the case, could we use `> /dev/null` or `of=/dev/null` to just be explicit about blackholing the data?

This might be slightly less alarming for anyone reading the code in future who fears we're about to fill their disk with garbage 😄.